File: server/internal/dto/wishlist.go
package dto

import (
	"time"

	"github.com/google/uuid"
)

type WishlistItemResponse struct {
	ID              uuid.UUID `json:"id"`
	VariantID       uuid.UUID `json:"variant_id"`
	ProductID       uuid.UUID `json:"product_id"`
	ProductName     string    `json:"product_name"`
	ProductSlug     string    `json:"product_slug"`
	VariantSize     string    `json:"variant_size"`
	VariantColor    string    `json:"variant_color"`
	ImageURL        string    `json:"image_url"`
	Price           float64   `json:"price"`
	AdditionalPrice float64   `json:"additional_price"`
	Stock           int       `json:"stock"`
	CreatedAt       time.Time `json:"created_at"`
}

type WishlistCheckResponse struct {
	InWishlist bool `json:"in_wishlist"`
}

type AddWishlistItemRequest struct {
	VariantID uuid.UUID `json:"variant_id" binding:"required"`
}

File: client/src/lib/api/wishlist.ts
import type { ApiResponse } from "@/types"
import { customerClient } from "./customerClient"

export type WishlistItem = {
  id: string
  variant_id: string
  product_id: string
  product_name: string
  product_slug: string
  variant_size: string
  variant_color: string
  image_url: string
  price: number
  additional_price: number
  stock: number
  created_at: string
}

export const wishlistApi = {
  getWishlist: async (): Promise<ApiResponse<WishlistItem[]>> => {
    const response = await customerClient.get<ApiResponse<WishlistItem[]>>("/wishlist")
    return response.data
  },

  checkWishlist: async (variantId: string): Promise<ApiResponse<{ in_wishlist: boolean }>> => {
    const response = await customerClient.get<ApiResponse<{ in_wishlist: boolean }>>(`/wishlist/check/${variantId}`)
    return response.data
  },

  addItem: async (variantId: string): Promise<ApiResponse<null>> => {
    const response = await customerClient.post<ApiResponse<null>>("/wishlist/items", { variant_id: variantId })
    return response.data
  },

  removeItem: async (variantId: string): Promise<ApiResponse<null>> => {
    const response = await customerClient.delete<ApiResponse<null>>(`/wishlist/items/${variantId}`)
    return response.data
  },
}
